flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java

```diff
@@ -610,7 +626,9 @@ public DataType getDataType(String key) {
 	public Optional<TableSchema> getOptionalTableSchema(String key) {
 		// filter for number of fields
 		final int fieldCount = properties.keySet().stream()
-			.filter((k) -> k.startsWith(key) && k.endsWith('.' + TABLE_SCHEMA_NAME))
+			.filter((k) -> k.startsWith(key)
+					// "key." is the prefix.
+					&& SCHEMA_COLUMN_NAME_SUFFIX.matcher(k.substring(key.length() + 1)).matches())
```

Comment:
We can just to exclude the primary key, then don't need the regex matching. 

```
.filter((k) -> k.startsWith(key) && !k.startsWith(key + "." + PRIMARY_KEY) && k.endsWith('.' + NAME))
```

Reply:
I think the regex matching is a more general way to match column names for the long run, just remove some specific keys seems hacky from my side.